### PR TITLE
Add one time trigger.

### PR DESCRIPTION
**Pull request by rizakii**
- Add ScheduleOnce predicate
- Add ScheduleOnce event source

warning : do not merge, this is ongoing task do not merge.

### CODE REVIEW

**Comment by rizakii** [comment]
@satoshikumano please review the API design.

**Comment by satoshikumano** [comment]
@rizakii LGTM.

**Comment by rizakii** [comment]
closed because need to address other issues.

**Comment by rizakii** [comment]
Related PR : 
- refactor Trigger #113 
- Android PR : https://github.com/KiiPlatform/thing-if-AndroidSDK/pull/81

**Comment by rizakii** [comment]
@satoshikumano please review the implementations.

File: ThingIFSDK/ThingIFSDK/Predicate.swift [hunk 1]
```diff
@@ -111,3 +115,44 @@ public class StatePredicate: NSObject,Predicate {
...
+    public let scheduleAt: NSDate
+
+    public func getEventSource() -> EventSource {
+        return EventSource.ScheduleOnce
+    }
+    /** Instantiate new ScheduleOncePredicate.
+
+     -Parameter scheduleAt: Specify execution schedule. It must be future date.
+     */
+    public init?(scheduleAt: NSDate) {
```
Comment (satoshikumano) [inline ThingIFSDK/ThingIFSDK/Predicate.swift]: Why Nullable?

File: ThingIFSDK/ThingIFSDK/Predicate.swift [hunk 2]
```diff
@@ -111,3 +115,44 @@ public class StatePredicate: NSObject,Predicate {
...
+        //TODO: implementations, please ignore
```
Comment (satoshikumano) [inline ThingIFSDK/ThingIFSDK/Predicate.swift]: Please remove the comment.

File: ThingIFSDK/ThingIFSDK/ThingIFAPI+Trigger.swift [hunk 3]
```diff
@@ -8,6 +8,26 @@
 
 import Foundation
 
+extension Predicate {
+    func validate() -> ThingIFError? {
```
Comment (satoshikumano) [inline ThingIFSDK/ThingIFSDK/ThingIFAPI+Trigger.swift]: Let's implement validate login in each classes.
(Would like to follow OOP standard)

**Comment by satoshikumano** [comment]
@rizakii please confirm.

**Comment by satoshikumano** [comment]
Test will be added in another PR.

**Pull request merged**